src/main/java/nextstep/subway/domain/Section.java

```diff
@@ -52,4 +52,42 @@ public Station getDownStation() {
     public int getDistance() {
         return distance;
     }
+
+    public void changeUpStation(Station station, int distance) {
+        upStation = station;
+        this.distance -= distance;
+    }
+
+    public void changeDownStation(Station station, int distance) {
+        downStation = station;
+        this.distance -= distance;
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
Distance정보는 달라도 동일한 엔티티로 취급하는 것일까요? Id는 또 어떨까요? 
이 equals 메서드가 충분히 동치관계를 보장할 수 있을까요? 
JPA Buddy에서 제공하는 equals and hashCode는 준영속 상태의 비교때문에 부족함이 있고, 또 다른 방안은 없을까요? 저는 대체로 IDE에서 제공하는 자동생성기능의 equals and hashCode를 추천드리는바에요 ㅎㅎ

```suggestion
    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof Section)) {
            return false;
        }
        Section section = (Section) o;
        return distance == section.distance 
                && Objects.equals(id, section.id) 
                && Objects.equals(line, section.line) 
                && Objects.equals(upStation, section.upStation) 
                && Objects.equals(downStation, section.downStation);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, line, upStation, downStation, distance);
    }
```